src/main/java/com/rackspace/jenkins_nodepool/NodePoolRequestStateWatcher.java

```diff
@@ -0,0 +1,127 @@
+            latch.countDown();
+            log.warning(e.getClass().getSimpleName() + " while registering watcher. Message: " +
+                    e.getLocalizedMessage());
+        }
+    }
+
+    /**
+     * Our event callback method.  When something changes and the request is fulfilled, we'll adjust our latch and the
+     * blocking call to waitUntilDone() will return.
+     *
+     * @param event the watch event value provided by the callback
+     */
+    @Override
+    public void process(WatchedEvent event) {
+        log.fine("Event received: " + event.toString());
+
+        // Only interested in the node data changed event
+        if (event.getType() == Event.EventType.NodeDataChanged) {
+            try {
+                // Let's fetch the data from ZK directly so we can read the state value
+                final RequestState state = getStateFromPath();
+
+                // Are we in the desired state?
+                if (state == desiredState) {
+                    latch.countDown();
+                } else {
+                    log.fine("Ignoring event type: " + event.getType() + " with state: " + state);
+                }
+            } catch (Exception e) {
+                log.warning(e.getClass().getSimpleName() +
+                        " occurred while updating state from Zookeeper. Message: " + e.getLocalizedMessage() +
+                        ". Ignoring the event.");
+            }
+        } else {
+            log.fine("Ignoring event type: " + event.getType());
+        }
+    }
+
+    /**
+     * Waits until event fires or until the specified timeout.
+     *
+     * @param timeout the timeout value
+     * @param unit    the unit of the timeout value, typically TimeUnit.SECONDS
+     * @return {@code true} if the wait returned before the timeout and {@code false}
+     * if the waiting time elapsed before the count reached zero
+     * @throws InterruptedException if the current thread is interrupted while waiting
+     */
+    public boolean waitUntilDone(long timeout, TimeUnit unit) throws InterruptedException {
+        return latch.await(timeout, unit);
+    }
+
+    /**
+     * Convenience routine to return the state value from the zpath.
+     *
+     * @return the RequestState value stored in the zpath.
+     * @throws KeeperException      if the ZooKeeper server signals an error with a non-zero error code
+     * @throws InterruptedException if the ZooKeeper server transaction is interrupted.
+     */
+    private RequestState getStateFromPath() throws KeeperException, InterruptedException {
+
+        // TODO: DAD - figure out how to use the NodeRequest object without all the extra stuff  (e.g. NodePool and Task)
+        // Let's fetch the data from ZK directly so we can read the state value
+
+        // Read the raw values
+        byte[] bytes = zk.getData(this.zpath, null, null);
+
+        // Convert to a string
+        String jsonString = new String(bytes, Charset.forName("UTF-8"));
```

Comment:
Should use NodePool.getCharset

Reply:
Unfortunately, I don't have a reference to NodePool in this class...and the `NodePool.getCharset()` method is not static (which is probably can be...)